src/Tests/Microsoft.NET.TestFramework/Commands/ShimCommand.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.TestFramework.Commands
+{
+    public class ShimCommand : TestCommand
+    {
+        private readonly string _commandPath;
+
+        public ShimCommand(ITestOutputHelper log, string commandPath, params string[] args) : base(log)
+        {
+            if (!File.Exists(commandPath))
+            {
+                throw new ArgumentException($"Cannot find command {commandPath}");
+            }
+
+            _commandPath = commandPath;
+            Arguments.AddRange(args);
+        }
+
+        protected override SdkCommandSpec CreateCommand(string[] args)
+        {
+            var sdkCommandSpec = new SdkCommandSpec()
+            {
+                FileName = _commandPath,
+                Arguments = args.ToList(),
+                WorkingDirectory = WorkingDirectory,
+            };
+
+
+            string dotnetRoot = Path.GetDirectoryName(TestContext.Current.ToolsetUnderTest.DotNetHostPath);
+            if (Environment.Is64BitProcess)
+            {
+                sdkCommandSpec.Environment.Add("DOTNET_ROOT", dotnetRoot);
+            }
+            else
+            {
+                sdkCommandSpec.Environment.Add("DOTNET_ROOT(x86)", dotnetRoot);
+            }
```

Comment:
Maybe we should do this for all tests here:  [https://github.com/dotnet/sdk/blob/f0ead5620a318ca57aa1a9e3e7b18db5d6984eaf/src/Tests/Microsoft.NET.TestFramework/ToolsetInfo.cs#L51](https://github.com/dotnet/sdk/blob/f0ead5620a318ca57aa1a9e3e7b18db5d6984eaf/src/Tests/Microsoft.NET.TestFramework/ToolsetInfo.cs#L51)

Reply:
fixed moved to AddTestEnvironmentVariables